### PR TITLE
Amend nondeterminism notes

### PR DESCRIPTION
**Pull request by t-vi**
include atomicAdd commentary as this is less well known

There is some discussion in #12207

Unfortunately, I cannot seem to get the ..include working in `_tensor_docs.py` and `_torch_docs.py`. I could use a hint for that.

### CODE REVIEW

File: docs/source/notes/randomness.rst [hunk 1]
```diff
@@ -22,12 +21,25 @@ CPU and CUDA)
...
 
+There are some PyTorch functions that use CUDA functions that can be a source
+of non-determinism. One class of such CUDA functions are atomic operations,
+in particular :attr:`atomicAdd`, where the order of parallel additions to the
+same value is undetermined and, for floating-point variables, a source of
+variance in the result. PyTorch functions that use :attr:`atomicAdd` in the forward
+include :meth:`torch.Tensor.index_add_`, :meth:`torch.Tensor.scatter_add_`,
+:meth:`torch.bincount`.
+A number of operations have backwards that use :attr:`atomicAdd`, in particular
+:meth:`torch.nn.functional.embedding`, :meth:`torch.nn.functional.embedding_bag`,
```
Comment (colesbury) [inline docs/source/notes/randomness.rst]: Embedding doesn't use atomicAdd, although EmbeddingBag does.

https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Embedding.cu

**Comment by t-vi** [comment]
If the other attributions seem OK, I would propose to add this.

File: docs/source/notes/randomness.rst [hunk 2]
```diff
@@ -22,12 +21,25 @@ CPU and CUDA)
     import torch
     torch.manual_seed(0)
 
+There are some PyTorch functions that use CUDA functions that can be a source
+of non-determinism. One class of such CUDA functions are atomic operations,
+in particular :attr:`atomicAdd`, where the order of parallel additions to the
```
Comment (SsnL) [inline docs/source/notes/randomness.rst]: I'm a bit confused. Isn't addition order-agnostic?

Comment (ezyang) [reply docs/source/notes/randomness.rst]: Not in floating point land :)

**Review by ssnl — APPROVED** [review]
LGTM assuming that the doc links work

**Comment by Naman-ntc** [comment]
These changes do not show up in the master docs, is it expected? @t-vi

**Comment by t-vi** [comment]
Insofar as the master docs are at a previous git revision and not instantly updated, yes.

**Pull request closed without merging**